test/rust_analyzer/rust_analyzer_test_runner.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+# Runs Bazel build commands over rustfmt rules, where some are expected
+# to fail.
+#
+# Can be run from anywhere within the rules_rust workspace.
+
+set -euo pipefail
+
+if [[ -z "${BUILD_WORKSPACE_DIRECTORY:-}" ]]; then
+    echo "This script should be run under Bazel"
+    exit 1
+fi
+
+PACKAGE_NAME="$1"
+if [[ -z "${PACKAGE_NAME:-}" ]]; then
+    echo "The first argument should be the package name of the test target"
+    exit 1
+fi
+
+function generate_workspace() {
+    local temp_dir="$(mktemp -d -t rules_rust_test_rust_analyzer-XXXXXXXXXX)"
+    local new_workspace="${temp_dir}/rules_rust_test_rust_analyzer"
+
+    mkdir -p "${new_workspace}"
+    cat << EOF > "${new_workspace}/WORKSPACE.bazel"
+workspace(name = "rules_rust_test_rust_analyzer")
+local_repository(
+    name = "rules_rust",
+    path = "${BUILD_WORKSPACE_DIRECTORY}",
+)
+load("@rules_rust//rust:repositories.bzl", "rust_repositories")
+rust_repositories(include_rustc_srcs = True)
+load("@rules_rust//tools/rust_analyzer:deps.bzl", "rust_analyzer_deps")
+rust_analyzer_deps()
+EOF
+
+cat << EOF > "${new_workspace}/.bazelrc"
+build --keep_going
+test --test_output=errors
+build:strict --aspects=@rules_rust//rust:defs.bzl%rustfmt_aspect
+build:strict --output_groups=+rustfmt_checks
+build:strict --aspects=@rules_rust//rust:defs.bzl%rust_clippy_aspect
+build:strict --output_groups=+clippy_checks
```

Comment:
Do you need the rustfmt / clippy stuff here?

Reply:
No but it's the only context that these targets would get rustfmt and clippy applied to them so wanted the integration test to be stricter by including them. Do you think they'd be problematic?

Reply:
Sounds good to me, but please mention this in a code comment, I think this is not obvious otherwise.

Reply:
Done